src/obu.rs

```diff
@@ -2462,25 +2462,14 @@ unsafe fn parse_obus(c: &mut Rav1dContext, r#in: &Rav1dData, global: bool) -> Ra
 
                         let country_code = country_code as u8;
                         let country_code_extension_byte = country_code_extension_byte as u8;
-                        // We need our public headers to be C++ compatible, so payload can't be
-                        // a flexible array member
-                        let payload = (*r#ref)
-                            .data
-                            .cast::<u8>()
-                            .offset(::core::mem::size_of::<DRav1d<Rav1dITUTT35, Dav1dITUTT35>>()
-                                as isize);
-                        let payload_size = payload_size as usize;
-                        for i in 0..payload_size {
-                            *payload.offset(i as isize) = gb.get_bits(8) as u8;
-                        }
+                        let payload = (0..payload_size).map(|_| gb.get_bits(8) as u8).collect(); // TODO(kkysen) fallible allocation
```

Comment:
Huh, I didn't realize you could collect into a `Box<[T]>`. I'm guessing it's just using a `Vec` in the middle, but it's nice that the stdlib APIs are there to make that work automatically

Reply:
I'm not sure what it's using in the middle; it could conceivably skip the overhead of an intermediary `Vec` with this API.  `.collect()` is so nice.